modules/tinymce/src/core/main/ts/api/EditorManager.ts

```diff
@@ -490,10 +490,19 @@ const EditorManager: EditorManager = {
     const initEditors = function () {
       let initCount = 0;
       const editors = [];
-      let targets: Element[];
+      let targets: HTMLElement[];
+
+      const createEditor = function (id: string, settings: RawEditorSettings, targetElm: HTMLElement) {
+        const targetElmWrap = SugarElement.fromDom(targetElm);
+        const tagSnapshot = Attr.clone(targetElmWrap);
 
-      const createEditor = function (id, settings, targetElm) {
         const editor: Editor = new Editor(id, settings, self);
+        editor.on('remove', () => {
+          Arr.each(Arr.map(targetElm.attributes, (attr) => attr.name), (name: string) => {
```

Comment:
This is looping twice when it's not needed. Additionally, the string type is not needed here/is an unnecessary type. So this could be simplified to this:
```
Arr.each(targetElm.attributes, (attr) => {
  Attr.remove(targetElmWrap, attr.name);
});
```

Reply:
Looping twice is needed so that we don't end up modifying the `attributes` list while mid-way through iterating through it*. That being said, I can rewrite it to make it more explicit that that's why the code is how it is.

*It was originally written the way that you have it there, and it'd bug out with some of the attributes in the list being `undefined` by the time the code got to them.

Reply:
That's not needed though, but you are correct about mutating the attributes being an issue. To solve that, you just need to loop backwards over the list, so you'd need to swap `each` to `eachr`.

Reply:
Change of plan I'm using `eachr` because that exists